Sources/SpeziFoundation/Concurrency/DiscardTaskGroup+CancellableTask.swift

```diff
@@ -0,0 +1,65 @@
+/// A handle to a child task that can be cancelled
+public struct CancelableTaskHandle: Sendable {
+    @usableFromInline private(set) var continuation: AsyncStream<Void>.Continuation
```

Comment:
does this need to be mutable? (given the fact that the continuation's `finish()` function is not mutating, and that we probably don't want a `CancelableTaskHandle` to possibly point to different tasks over the course of its lifetime)

Reply:
great catch 👍